flatland_viz/src/pause_sim_tool.cpp

```diff
@@ -5,19 +5,20 @@ namespace flatland_viz {
 PauseSimTool::PauseSimTool() {}
 
 // Disconnect the service client when the tool's destructor is called
-PauseSimTool::~PauseSimTool() { pause_service_.shutdown(); }
+PauseSimTool::~PauseSimTool() { rclcpp::shutdown(); /*pause_service_->shutdown();*/ }
 
 // When the tool is initially loaded, connect to the pause toggle service
 void PauseSimTool::onInitialize() {
-  pause_service_ = nh_.serviceClient<std_srvs::Empty>("toggle_pause");
-  setName("Pause/Resume");
+    std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("pause_sim_tool");  // TODO
```

Comment:
Not sure if it is a good way to create the node locally here or better as member variable?